src/data_provider/src/sysInfoWin.cpp

```diff
@@ -756,9 +752,24 @@ nlohmann::json SysInfo::getPorts() const
     return ports;
 }
 
-void SysInfo::getProcessesInfo(std::function<void(nlohmann::json&)> /*callback*/) const
+void SysInfo::getProcessesInfo(std::function<void(nlohmann::json&)> callback) const
 {
-    // TO DO
+    fillProcessesData([&callback](const auto &processEntry)
+    {
+        auto processInfo { getProcessInfo(processEntry) };
```

Comment:
You should change this by using copy assingment instead. We've had issues with this in the past.
Also, by doing this, you can remove the code from line 763 to 770 and just leave the callback call